pkg/core/bootstrap/bootstrap.go

```diff
@@ -160,21 +160,14 @@ func startReporter(runtime core_runtime.Runtime) error {
 }
 
 func initializeBootstrap(cfg kuma_cp.Config, builder *core_runtime.Builder) error {
-	var pluginName core_plugins.PluginName
-	switch cfg.Environment {
-	case config_core.KubernetesEnvironment:
-		pluginName = core_plugins.Kubernetes
-	case config_core.UniversalEnvironment:
-		pluginName = core_plugins.Universal
-	default:
-		return errors.Errorf("unknown environment type %s", cfg.Environment)
-	}
-	plugin, err := core_plugins.Plugins().Bootstrap(pluginName)
-	if err != nil {
-		return errors.Wrapf(err, "could not retrieve bootstrap %s plugin", pluginName)
-	}
-	if err := plugin.Bootstrap(builder, nil); err != nil {
-		return err
+	for name, plugin := range core_plugins.Plugins().BootstrapPlugins() {
+		if (cfg.Environment == config_core.KubernetesEnvironment && name == core_plugins.Universal) ||
+			(cfg.Environment == config_core.UniversalEnvironment && name == core_plugins.Kubernetes) {
+			continue
```

Comment:
why it is not an error or panic?

Reply:
Because we go through all plugins. We want to skip kubernetes plugin on universal and universal plugin on kubernetes. Probably a better way to do it would be to have a list of plugins to load and refactor all plugins to load in such way, but I don't have time for it for now